### PR TITLE
OAuth model and service added, tests to cover the newly added methods

### PR DESCRIPTION
**Pull request by SanchitaJain4**

### CODE REVIEW

File: backend/chromeApiServer/src/main/java/com/canvas/controllers/chromeApiServer/ChromeApiController.java [hunk 1]
```diff
@@ -127,6 +133,51 @@ public ResponseEntity<CommandOutput> initiateStudentCodeEvaluation(
...
+     * @return Redirects to loginSuccess/loginFailure route based on access token response from canvas
+     * @throws IncorrectRequestParamsException
+     */
+    @GetMapping(
+            value = "/oauth2Response"
+    )
+    public ResponseEntity<String> oAuth2Response (
+            @RequestParam(value = "code", required = true) String code) {
+        try {
+            String accessToken = this.oauthService.fetchAccessTokenResponse(code,"http://localhost:8080/oauth2Response");
```
Comment (SanchitaJain4) [inline backend/chromeApiServer/src/main/java/com/canvas/controllers/chromeApiServer/ChromeApiController.java:149]: The URL needs to be changed to our backend server. 
@fkogge / @gaaliciA1990  could you point me to the right URL?

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import okhttp3.Response;
+
+/**
+ * This class handles OAuth requests for generation, validation of access tokens
+ */
+public class OAuthService {
+
+    private final CanvasClientService canvasClientService;
+    //TODO: Add and fetch values from config
```
Comment (fkogge) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:13]: should we implement this TODO before merging to master?

Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:13]: I will do it in a separate task, where I will remove such constants and store it in a config file.

File: backend/chromeApiServer/src/main/java/com/canvas/controllers/chromeApiServer/ChromeApiController.java (hunk 1 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/controllers/chromeApiServer/ChromeApiController.java]: This is updated.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java (hunk 2 above)
Comment (gaaliciA1990) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:13]: If this is part of the Oauth process, this should be implemented before merging master. Please fix before we can approve this PR

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java [hunk 3]
```diff
@@ -319,7 +354,7 @@ private CanvasAPIException throwCanvasException(Exception e) {
      * @return JsonNode
      * @throws IOException error message thrown
      */
-    protected JsonNode parseResponseToJsonNode(Response response) throws IOException {
+    public static JsonNode parseResponseToJsonNode(Response response) throws IOException {
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java:357]: Why did you move this to public? We made it protected since it is a helper function that shouldn't be accessible to other classes.

**Review by gaaliciA1990 — CHANGES_REQUESTED** [review]
Please address the comments before this can be merged into the main branch

**Review by gaaliciA1990 — APPROVED** [review]
